crates/ruff_python_formatter/src/expression/parentheses.rs

```diff
@@ -110,26 +110,57 @@ where
 {
     FormatParenthesized {
         left,
+        comments: &[],
+        content: Argument::new(content),
+        right,
+    }
+}
+
+/// Formats `content` enclosed by the `left` and `right` parentheses, along with any dangling
+/// comments that on the parentheses themselves.
+pub(crate) fn parenthesized_with_dangling_comments<'content, 'ast, Content>(
```

Comment:
Nit: We so far preferred to have `with_*` methods on the builders rather than having many dedicated builder methods

```rust
parenthesized("(", &expr.format(), ")").with_dangling_comments(comments)
```